### PR TITLE
JBIDE-26502 - Add support for Wildfly 15

### PR DESCRIPTION
**Pull request by jeffmaury**
Signed-off-by: Jeff MAURY <jmaury@redhat.com>

### CODE REVIEW

File: as/plugins/org.jboss.ide.eclipse.as.ui/plugin.xml [hunk 1]
```diff
@@ -444,9 +444,10 @@
 						<test property="org.eclipse.wst.server.ui.serverType" value="org.jboss.ide.eclipse.as.wildfly.90"></test>
 						<test property="org.eclipse.wst.server.ui.serverType" value="org.jboss.ide.eclipse.as.wildfly.100"></test>
 						<test property="org.eclipse.wst.server.ui.serverType" value="org.jboss.ide.eclipse.as.wildfly.110"></test>
-						<test property="org.eclipse.wst.server.ui.serverType" value="org.jboss.ide.eclipse.as.wildfly.120"></test>
+                        <test property="org.eclipse.wst.server.ui.serverType" value="org.jboss.ide.eclipse.as.wildfly.120"></test>
```
Comment (mpittkin) [inline as/plugins/org.jboss.ide.eclipse.as.ui/plugin.xml]: It's a minor quibble but is there a reason these can't be indented to match the rest?

File: as/plugins/org.jboss.ide.eclipse.as.wtp.core/src/org/jboss/ide/eclipse/as/core/server/bean/ServerBeanTypeWildfly140.java [hunk 2]
```diff
@@ -22,15 +22,15 @@ public ServerBeanTypeWildfly140() {
...
+				new String[]{"14.0"}, new Wildfly140ServerTypeCondition());
 	}
 	
 	@Override
 	protected String getServerTypeBaseName() {
 		return getId();
 	}
 	
-	public static class Wildfly130ServerTypeCondition extends AbstractCondition {
+	public static class Wildfly140ServerTypeCondition extends AbstractCondition {
```
Comment (mpittkin) [inline as/plugins/org.jboss.ide.eclipse.as.wtp.core/src/org/jboss/ide/eclipse/as/core/server/bean/ServerBeanTypeWildfly140.java]: Good catch

File: as/itests/org.jboss.tools.as.itests/src/org/jboss/tools/as/itests/EJB3SupportVerifierTest.java [hunk 3]
```diff
@@ -71,6 +71,7 @@
 		expected.put(IJBossToolingConstants.SERVER_EAP_61, true);
 		expected.put(IJBossToolingConstants.SERVER_EAP_70, true);
 		expected.put(IJBossToolingConstants.SERVER_EAP_71, true);
+		expected.put(IJBossToolingConstants.SERVER_WILDFLY_150, true);
```
Comment (robstryker) [inline as/itests/org.jboss.tools.as.itests/src/org/jboss/tools/as/itests/EJB3SupportVerifierTest.java]: Out of order. Put WF15 below 14?

File: as/plugins/org.jboss.ide.eclipse.as.core/jbosscore/org/jboss/ide/eclipse/as/core/server/internal/extendedproperties/Wildfly140ExtendedProperties.java [hunk 4]
```diff
@@ -61,13 +61,4 @@ public boolean allowExplodedModulesInWarLibs() {
 	public boolean allowExplodedModulesInEars() {
 		return true;
 	}
-	
-
-	/**
-	 * Wildfly 11 appears to work through java 9 to varying degrees
```
[the remark this answers is not in the document]
Comment (jeffmaury) [reply as/plugins/org.jboss.ide.eclipse.as.core/jbosscore/org/jboss/ide/eclipse/as/core/server/internal/extendedproperties/Wildfly140ExtendedProperties.java]: No I removed null which now means by default max=Java8 so we're safe. But I will check if either WF11, 12, 13 or 14 can be run with Java9+ on report the change I've done for WF15

File: as/itests/org.jboss.tools.as.itests/src/org/jboss/tools/as/itests/EJB3SupportVerifierTest.java (hunk 3 above)
Comment (jeffmaury) [reply as/itests/org.jboss.tools.as.itests/src/org/jboss/tools/as/itests/EJB3SupportVerifierTest.java]: That was a side effect of your script :)

**Comment by robstryker** [comment]
Build is failing due to some sonarcloud stuff. I don't know who can look into that, but I'm not familiar with that stuff at all. 

Rest of the build looked good though.

**Pull request merged**